targetcli/ui_backstore.py

```diff
@@ -423,17 +423,17 @@ def _create_file(self, filename, size, sparse=True):
             raise ExecutionError("Could not open %s" % filename)
         try:
             if sparse:
+                os.ftruncate(f.fileno(), size)
+            else:
+                self.shell.log.info("Writing %d bytes" % size)
                 try:
                     os.posix_fallocate(f.fileno(), 0, size)
                 except AttributeError:
                     # Prior to version 3.3, Python does not provide fallocate
-                    os.ftruncate(f.fileno(), size)
-            else:
-                self.shell.log.info("Writing %d bytes" % size)
-                while size > 0:
-                    write_size = min(size, 1024)
-                    f.write("\0" * write_size)
-                    size -= write_size
+                    while size > 0:
+                        write_size = min(size, 1024)
```

Comment:
Do we only accept size in 1024 chucks only?
Say what if the user asks to create a storage object with a size aligned to 512? say request to create for 2560B?

Reply:
@pkalever No, it calls min() so with your example the while() will do 3 cycles:
two cycles with write_size = 1024 and the last one with write_size = min(512, 1024) = 512

Reply:
perfect!